mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergPartitionTextParser.java

```diff
@@ -0,0 +1,123 @@
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.types.Types;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.apache.iceberg.types.Types.NestedField.optional;
+
+public class TestHiveIcebergPartitionTextParser {
+
+  private static final Schema SCHEMA = new Schema(
+      optional(1, "id", Types.LongType.get()),
+      optional(2, "name", Types.StringType.get()),
+      optional(3, "employee_info", Types.StructType.of(
+          optional(6, "employer", Types.StringType.get()),
+          optional(7, "id", Types.LongType.get()),
+          optional(8, "address", Types.StringType.get())
+      )),
+      optional(4, "created", Types.TimestampType.withoutZone()),
+      optional(5, "updated", Types.TimestampType.withoutZone())
+  );
+
+  @Test
+  public void testParsingColumns() {
+    PartitionSpec expected = PartitionSpec.builderFor(SCHEMA)
+        .identity("id").identity("name").build();
+    Assert.assertEquals(expected, HiveIcebergPartitionTextParser.fromText(SCHEMA, "id|name"));
+    Assert.assertEquals("space on the right of pipe should not matter",
+        expected, HiveIcebergPartitionTextParser.fromText(SCHEMA, "id| name"));
+    Assert.assertEquals("space on the left of pipe should not matter",
+        expected, HiveIcebergPartitionTextParser.fromText(SCHEMA, "id |name"));
+    Assert.assertEquals("space on both sides of pipe should not matter",
+        expected, HiveIcebergPartitionTextParser.fromText(SCHEMA, "id | name"));
+  }
+
+  @Test
+  public void testParsingColumnsFailure() {
+    AssertHelpers.assertThrows("should fail when column does not exist",
+        ValidationException.class,
+        "Cannot find column",
+        () -> HiveIcebergPartitionTextParser.fromText(SCHEMA, "col"));
+  }
+
+  @Test
+  public void testParsing2ArgTransforms() {
+    PartitionSpec expected = PartitionSpec.builderFor(SCHEMA)
+        .bucket("name", 16).truncate("employee_info.address", 8).build();
+    Assert.assertEquals(expected, HiveIcebergPartitionTextParser.fromText(
+        SCHEMA, "bucket(name,16)|truncate(employee_info.address,8)"));
+    Assert.assertEquals("space in args should not matter",
+        expected, HiveIcebergPartitionTextParser.fromText(
+            SCHEMA, "bucket( name, 16)| truncate(employee_info.address ,8 )"));
+  }
+
+  @Test
+  public void testParsing2ArgTransformsFailure() {
+    AssertHelpers.assertThrows("should fail when column does not exist",
+        ValidationException.class,
+        "Cannot find column",
+        () -> HiveIcebergPartitionTextParser.fromText(SCHEMA, "bucket(col, 8)"));
+
+    AssertHelpers.assertThrows("should fail when input to transform is wrong",
+        ValidationException.class,
+        "Cannot parse 2-arg partition transform from text part",
+        () -> HiveIcebergPartitionTextParser.fromText(SCHEMA, "bucket(8, name)"));
```

Comment:
Nit: could add cases like `bucket(name)` for 2-args transforms, and `day(1, created)` for 1-arg transform